### PR TITLE
feat(anta): Added the test case to verify the Entropy source security 

### LINKED ISSUE #768: Add the test case to verify the Entropy source security
**Pass/Fail Criteria:**

Hardware entropy generation should be enabled

### PR DESCRIPTION
**Pull request by vitthalmagadum**
# Description

Verifies hardware entropy generation is enabled on device.

    Expected Results
    ----------------
    * Success: The test will pass if hardware entropy generation is enabled.
    * Failure: The test will fail if hardware entropy generation is not enabled.

Fixes #768 

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have run pre-commit for code linting and typing (`pre-commit run`)
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

File: anta/tests/security.py [hunk 1]
```diff
@@ -820,3 +820,41 @@ def test(self) -> None:
...
+    Examples
+    --------
+    ```yaml
+    anta.tests.security:
+      - VerifyHardwareEntropy:
+    ```
+    """
+
+    name = "VerifyHardwareEntropy"
+    description = "Verifies hardware entropy generation."
```
Comment (MaheshGSLAB) [inline anta/tests/security.py:843]:
```suggestion
    description = "Verifies hardware entropy generation is enabled on device."
```

File: anta/tests/security.py [hunk 2]
```diff
@@ -820,3 +820,41 @@ def test(self) -> None:
...
+
+        command_output = self.instance_commands[0].json_output
+
+        # Check if security parameters are configured
+        if not command_output:
+            self.result.is_failure("No cryptographic algorithm is configured.")
+            return
+
+        if not command_output.get("hardwareEntropyEnabled"):
+            self.result.is_failure("Hardware entropy generation is disabled.")
```
Comment (MaheshGSLAB) [inline anta/tests/security.py:860]:
```suggestion

        command_output = self.instance_commands[0].json_output

        # Check if security parameters are configured
        if not command_output:
            self.result.is_failure("No cryptographic algorithm is configured.")
            return

        if not command_output.get("hardwareEntropyEnabled"):
            self.result.is_failure("Hardware entropy generation is disabled.")
        else:
            self.result.is_success()
```

File: anta/tests/security.py [hunk 3]
```diff
@@ -820,3 +820,41 @@ def test(self) -> None:
...
+    @AntaTest.anta_test
+    def test(self) -> None:
+        """Main test function for VerifyHardwareEntropy."""
+        self.result.is_success()
+
+        command_output = self.instance_commands[0].json_output
+
+        # Check if security parameters are configured
+        if not command_output:
+            self.result.is_failure("No cryptographic algorithm is configured.")
```
Comment (MaheshGSLAB) [inline anta/tests/security.py:856]: Do we have this case that command reflect an empty output? If yes then its ok

Comment (vitthalmagadum) [reply anta/tests/security.py]: IMO, We don't have such case.
removing the check.
Thanks!

File: anta/tests/security.py (hunk 2 above)
Comment (vitthalmagadum) [reply anta/tests/security.py]: Handled

File: anta/tests/security.py (hunk 1 above)
Comment (vitthalmagadum) [reply anta/tests/security.py]: Handled

**Review by MaheshGSLAB — APPROVED** [review]
LGTM

**Review by carl-baillargeon — APPROVED** [review]
LGTM

**Pull request merged**